### ISSUE TITLE
Maskzero breaks with cuda

### ISSUE THREAD

**Issue by RicherMans** [Element-Research/rnn#313]
Hey there, I recently updated my torch and rnn( including dpnn, torchx ) and after updating the maskzero layer does not seem to work. The following code demonstrates the problem:

```lua 
require 'rnn'
require 'nn'

cuda = arg[1]

mdl = nn.Sequential()

mdl:add(nn.MaskZero(nn.Linear(10,20),1))
mdl:add(nn.ReLU())
mdl:add(nn.LogSoftMax())
inp = torch.rand(10,10)
if cuda == "cuda" then
	require 'cunn'
	mdl = mdl:cuda()
	inp = inp:cuda()
end
print(mdl:forward(inp))
```

When I run this script on the cpu, it works without any problem, but if I use it on the GPU, it crashes with:
```
MaskZero.lua:49: invalid arguments: CudaTensor CudaTensor number 
expected arguments: *CudaTensor* CudaByteTensor float
```

**Comment by nicholas-leonard** [comment]
@RicherMans Good catch. Apparently, this cutorch commit https://github.com/torch/cutorch/commit/20001aca5b7e49c414011938a2d79637924f2888 breaks backwards compatibility. I am looking into it.

**Comment by nicholas-leonard** [comment]
fixed by #319
